app/questionnaire/rules.py

```diff
@@ -36,6 +38,53 @@ def evaluate_rule(when, answer_value):
     return result
 
 
+def evaluate_date_rule(when, answer_store, group_instance, metadata, answer_value):
+    result = False
+
+    match_value = None
+    date_comparison = when['date_comparison']
+
+    if 'value' in date_comparison:
+        if date_comparison['value'] == 'now':
+            match_value = datetime.utcnow().strftime('%Y-%m-%d')
+        else:
+            match_value = date_comparison['value']
+    elif 'id' in date_comparison:
+        match_value = get_answer_store_value(date_comparison['id'], answer_store, group_instance)
+    elif 'meta' in date_comparison:
+        match_value = get_metadata_value(metadata, date_comparison['meta'])
+
+    match_value = convert_to_datetime(match_value)
+
+    if 'offset_by' in date_comparison and match_value:
```

Comment:
If `match_value` is `None` then this will error if the code tries to compare it against `answer_value` using `<` or `>` operators.

Consider returning immediately if `match_value` is `None`.